agent/config/config.go

```diff
@@ -249,7 +252,9 @@ func environmentConfig() Config {
 	// invalid parse
 	// Blank is not a warning; we have sane defaults
 	if err != io.EOF && err != nil {
-		seelog.Warnf("Invalid format for \"ECS_RESERVED_PORTS_UDP\" environment variable; expected a JSON array like [1,2,3]. err %v", err)
+		msg := fmt.Sprintf("Invalid format for \"ECS_RESERVED_PORTS_UDP\" environment variable; expected a JSON array like [1,2,3]. err %v", err)
+		seelog.Warn(msg)
+		errs = append(errs, errors.New(string(msg)))
```

Comment:
That's correct. This change could cause configuration that was previously accepted (albeit ignored) by the agent to now become a fatal error. I assert that this is desirable. Ignoring configuration errors and running anyway is rarely desired and is almost certainly not what the customer expected us to do. It's unlikely that the customer is looking at the logs when the system appears to be operating as intended, and by the time the system stops operating as intended, they may be in the midst of what should have been a preventable service disruption.

I could be persuaded that this is not the right place to be introducing these changes, but I do believe we should make them sooner or later.